### DISCUSSION TITLE
Question: pomerium cache cointainer refresh time

### DISCUSSION THREAD

**Discussion by Mberga14** [pomerium/pomerium#1625] [General]
Hello, I have a question regarding the cache and databroker containers. 
We currently have an in-memory setup and sometimes when we add people to groups it takes more than 24 hours for the cache to refresh. We have a workaround (we restart the cache, and it works instantly) but we would like to avoid restarting containers. My question is, how much time does it take for cache service to sync groups and if there is a variable that we could set so this would happen sooner? 

Looking forward to an answer,

Regards

**Comment by desimone** [comment]
(Converted this issue to a discussion) 

@Mberga14  ,

Can you tell me a bit more your configuration? Which identity provider are you using and so on? What version? Can you tell me a bit more about your environment? 

Typically -- updates should be pretty frequent, and the [default is 30 minutes](https://www.pomerium.io/reference/#identity-provider-refresh-directory-settings).

**Reply by Mberga14** [reply]
We are running  pomerium tag `v0.11.0-rc2` in our k8s cluster that is hosted on Azure. We are using azure provider.  We have it setup as documentation states [hhere](https://www.pomerium.io/docs/identity-providers/azure.html) (without the optional service account). The interesting this is, if we reset the cache cointainer, changes start being applied immidiately.  I will try to edit the `IDP_REFRESH_DIRECTORY_INTERVAL` variable and report back to you.

Thanks!

**Reply by Mberga14** [reply]
Hey @desimone ! 
I finally got around to testing this, and settign the environment variables fixed the issue for me. I suspect the variables default setting might not be working as intended, as when I do not set those variables manually, the cache does not refresh on the standard interval. After setting it to `IDP_REFRESH_DIRECTORY_INTERVAL=30m` and  `IDP_REFRESH_DIRECTORY_TIMEOUT=10m` the cache started refreshing at the set interval.
